Waste.API/Services/WasteService.cs

```diff
@@ -143,5 +166,18 @@ private async Task<WasteJourney> GetJourney(int id)
         {
             return await _wasteRepository.GetById<WasteJourney>(id);
         }
+
+        public async Task SaveBaledWithWire(int journeyId, bool baledWithWire)
+        {
+            var journeyRecord = await GetJourney(journeyId);
+            if (journeyRecord == null)
+                throw new NullReferenceException(nameof(journeyRecord));
+
+            journeyRecord.BaledWithWire = baledWithWire;
+            if (journeyRecord.BaledWithWire == true)
+                journeyRecord.DeductionAmount = _deductionAmount;
```

Comment:
No need to compare a boolean
if (journeyRecord.BaledWithWire == true)

just use
if (journeyRecord.BaledWithWire)

Reply:
if (journeyRecord.BaledWithWire) will anyway return false if it is false or null